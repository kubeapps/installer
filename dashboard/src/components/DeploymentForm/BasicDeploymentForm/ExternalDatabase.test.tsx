import { shallow } from "enzyme";
import * as React from "react";

import ExternalDatabaseSection, {
  EXTERNAL_DB_HOST_PARAM_NAME,
  EXTERNAL_DB_PARAM_NAME,
  EXTERNAL_DB_PASSWORD_PARAM_NAME,
  EXTERNAL_DB_PORT_PARAM_NAME,
  EXTERNAL_DB_USER_PARAM_NAME,
  USE_SELF_HOSTED_DB_PARAM_NAME,
} from "./ExternalDatabase";

const defaultProps = {
  label: "Enable an external database",
  externalDatabaseParams: {
    [EXTERNAL_DB_PARAM_NAME]: { path: "edbs", value: {}, type: "object" },
    [EXTERNAL_DB_HOST_PARAM_NAME]: { path: "edbs.host", value: "localhost", type: "string" },
    [EXTERNAL_DB_USER_PARAM_NAME]: { path: "edbs.user", value: "user", type: "string" },
    [EXTERNAL_DB_PASSWORD_PARAM_NAME]: { path: "edbs.pass", value: "pass123", type: "string" },
    [EXTERNAL_DB_PORT_PARAM_NAME]: { path: "edbs.port", value: 1234, type: "integer" },
    [USE_SELF_HOSTED_DB_PARAM_NAME]: { path: "mariadb.enabled", value: true, type: "boolean" },
  },
  handleBasicFormParamChange: jest.fn(),
};

it("should render a external database section", () => {
  const wrapper = shallow(<ExternalDatabaseSection {...defaultProps} />);
  expect(wrapper).toMatchSnapshot();
});

it("should hide/show the database params if the self-hosted database is enabled/disabled", () => {
  const wrapper = shallow(<ExternalDatabaseSection {...defaultProps} />);
  expect(defaultProps.externalDatabaseParams[USE_SELF_HOSTED_DB_PARAM_NAME].value).toBe(true);
  expect(wrapper.find(".margin-t-normal").prop("hidden")).toBe(true);

  wrapper.setProps({
    ...defaultProps,
    externalDatabaseParams: {
      ...defaultProps.externalDatabaseParams,
      [USE_SELF_HOSTED_DB_PARAM_NAME]: { path: "mariadb.enabled", value: false, type: "boolean" },
    },
  });
  wrapper.update();
  expect(wrapper.find(".margin-t-normal").prop("hidden")).toBe(false);
});
