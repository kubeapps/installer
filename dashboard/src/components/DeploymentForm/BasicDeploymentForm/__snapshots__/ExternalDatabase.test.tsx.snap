// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a external database section 1`] = `
<div
  className="subsection margin-v-normal"
>
  <BooleanParam
    handleBasicFormParamChange={[Function]}
    id="enable-self-hosted-db"
    label="Use a Self Hosted Database"
    name="useSelfHostedDatabase"
    param={
      Object {
        "path": "mariadb.enabled",
        "type": "boolean",
        "value": true,
      }
    }
  />
  <div
    className="margin-t-normal"
    hidden={true}
  >
    <span>
      Enable an external database
    </span>
    <TextParam
      handleBasicFormParamChange={[Function]}
      id="externalDatabase-0"
      inputType="text"
      key="externalDatabase-0"
      label=""
      name="externalDatabase"
      param={
        Object {
          "path": "edbs",
          "type": "object",
          "value": Object {},
        }
      }
    />
    <TextParam
      handleBasicFormParamChange={[Function]}
      id="externalDatabaseHost-1"
      inputType="text"
      key="externalDatabaseHost-1"
      label="Host"
      name="externalDatabaseHost"
      param={
        Object {
          "path": "edbs.host",
          "type": "string",
          "value": "localhost",
        }
      }
    />
    <TextParam
      handleBasicFormParamChange={[Function]}
      id="externalDatabaseUser-2"
      inputType="text"
      key="externalDatabaseUser-2"
      label="User"
      name="externalDatabaseUser"
      param={
        Object {
          "path": "edbs.user",
          "type": "string",
          "value": "user",
        }
      }
    />
    <TextParam
      handleBasicFormParamChange={[Function]}
      id="externalDatabasePassword-3"
      inputType="text"
      key="externalDatabasePassword-3"
      label="Password"
      name="externalDatabasePassword"
      param={
        Object {
          "path": "edbs.pass",
          "type": "string",
          "value": "pass123",
        }
      }
    />
    <TextParam
      handleBasicFormParamChange={[Function]}
      id="externalDatabasePort-4"
      inputType="number"
      key="externalDatabasePort-4"
      label="Port"
      name="externalDatabasePort"
      param={
        Object {
          "path": "edbs.port",
          "type": "integer",
          "value": 1234,
        }
      }
    />
  </div>
</div>
`;
